src/pages/strategies/forms/CodeForm.js

```diff
@@ -0,0 +1,141 @@
+import { useState, useContext } from 'react'
+import { useFormik } from 'formik'
+import { Grid } from '@mui/material'
+import FormShell from 'src/components/Shared/FormShell'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { RequestsContext } from 'src/providers/RequestsContext'
+
+import { useEffect } from 'react'
+
+import * as yup from 'yup'
+import { DocumentReleaseRepository } from 'src/repositories/DocumentReleaseRepository'
+
+import toast from 'react-hot-toast'
+
+const CodeForm = ({
+  labels,
+  editMode,
+  maxAccess,
+  setEditMode,
+  recordId,
+  store,
+  setRefresh,
+  strategiesFormik,
+  window
+}) => {
+  const { postRequest, getRequest } = useContext(RequestsContext)
+
+  const { recordId: stgId } = store
+
+  const [initialValues, setInitialData] = useState({
+    codeId: '',
+    groupId: strategiesFormik.values.groupId,
+    strategyId: stgId
+  })
+
+  const formik = useFormik({
+    enableReinitialize: true,
+    validateOnChange: true,
+
+    initialValues,
+    validationSchema: yup.object({
+      codeId: yup.string().required()
+    }),
+    onSubmit: async (values, { setSubmitting }) => {
+      try {
+        await postGroups(values)
+        window.close()
+      } catch (error) {
+        toast.error('An error occurred')
+        setSubmitting(false)
```

Comment:
Why do we need this? Was this code copied from somewhere in the codebase or chatgpt'd?

Reply:
the code Form should be closed after submit